cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/RepositoryManager.java

```diff
@@ -79,20 +83,26 @@ public Path getBasePath() {
   public static void validateConfig(SecureStore secureStore, SourceControlConfig sourceControlConfig) throws
     RepositoryConfigValidationException {
     RepositoryConfig config = sourceControlConfig.getRepositoryConfig();
+    CredentialsProvider credentialsProvider = sourceControlConfig.getAuthenticationStrategy()
+      .getCredentialsProvider(secureStore, config, sourceControlConfig.getNamespaceID());
     // Try fetching heads in the remote repository.
     try (Git git = Git.wrap(new InMemoryRepository.Builder().build())) {
-      LsRemoteCommand cmd = createCommand(git::lsRemote, sourceControlConfig, secureStore).setRemote(config.getLink())
-                                                                                          .setHeads(true)
-                                                                                          .setTags(false);
+      LsRemoteCommand cmd =
+        createCommand(git::lsRemote, sourceControlConfig, credentialsProvider).setRemote(config.getLink())
+          .setHeads(true)
+          .setTags(false);
       validateDefaultBranch(cmd.callAsMap(), config.getDefaultBranch());
     } catch (TransportException e) {
       throw new RepositoryConfigValidationException(
         "Failed to connect with repository. Authentication credentials invalid or the repository may not exist.",
```

Comment:
should be `Authentication credentials may be invalid or the repository may not exist.`

Also, is the TransportException message not detailed enough? Invalid credentials or a non-existent repository are very different scenarios.

Reply:
Both errors result in TransportException, but the message is different. I've changed the message to "Failed to connect with repository" and appended the message from the Exception.